datafusion/core/src/datasource/file_format/csv.rs

```diff
@@ -298,7 +301,12 @@ impl CsvFormat {
 
         while let Some(chunk) = stream.next().await.transpose()? {
             let format = arrow::csv::reader::Format::default()
-                .with_header(self.options.has_header && first_chunk)
+                .with_header(
+                    self.options
+                        .has_header
+                        .unwrap_or(state.config_options().catalog.has_header)
+                        && first_chunk,
```

Comment:
It might be possible to avoid some redundant code (and also document the behavior) by pulling this into a function like

```rust
impl CsvFormat { 
..
  /// returns true if the first line should be treated like a header
  /// if not explicitly specified by the `CREATE TABLE` statement or format options
  /// defaults to the default `catalog.has_header` value
  fn has_header(&self) -> bool {
    self.options
      .has_header
      .unwrap_or(state.config_options().catalog.has_header)
  }
..
}
```